### PR TITLE
Fix panic strings.

### PR DESCRIPTION
**Pull request by dlrobertson**
- Fix panic string in `check_ast_crate`.
 - Update panic string for Duration subtraction on overflow/underflow.

Not sure if the changes to `Duration` are helpful/needed. Mostly just a nit.
Otherwise this is just a one character change :)

On another note: I hit the panic in `check_ast_crate` when compiling
[m-labs/smoltcp] with the following:

```
cargo test --doc --no-default-features --features "std socket-raw"`
```

[m-labs/smoltcp]: https://github.com/m-labs/smoltcp

### CODE REVIEW

File: src/libstd/time/duration.rs [hunk 1]
```diff
@@ -290,7 +290,7 @@ impl Duration {
     }
 
     /// Checked `Duration` subtraction. Computes `self - other`, returning [`None`]
-    /// if the result would be negative or if overflow occurred.
+    /// if underflow or overflow occurred.
```
Comment (kennytm) [inline src/libstd/time/duration.rs]: Please revert these changes. The term "overflow" is correct, see #46898.

Comment (dlrobertson) [reply src/libstd/time/duration.rs]: :+1:

**Comment by dlrobertson** [comment]
Updated! Such a massive PR lol

**Comment by kennytm** [comment]
Thanks!

@bors r+ rollup

**Comment by shepmaster** [comment]
> Such a massive PR lol

On the flip side, a lot of times the smallest changes are the ones you have to scrutinize the hardest. It's nice to have easier ones every so often ;-)
